### ISSUE TITLE
Question about the comparison of data augmentations's reproduce

### ISSUE THREAD

**Issue by tinaa4869** [princeton-nlp/SimCSE#281]
Hi, thanks for your great work!

I'm trying to reproduce the result about Table 1(Comparison of data augmentations):
<img width="116" alt="image" src="https://github.com/princeton-nlp/SimCSE/assets/129758008/32f8a513-d1db-43bd-814a-1363d32e8af8">

Since it's evaluated on STS-B development set, I think from the training process I can get the result.

For unsup SimCSE, I got the result:
<img width="148" alt="image" src="https://github.com/princeton-nlp/SimCSE/assets/129758008/bd2ae79e-313b-48cd-987f-19b3fb226764">
the STS-B spearman's correlation 0.8213 close to the results in the table.

Then I write a function to perform Crop:
<img width="240" alt="image" src="https://github.com/princeton-nlp/SimCSE/assets/129758008/3eb089e8-6a17-4da8-8825-c76bb8dd0bee">
And get the processed data by:
<img width="277" alt="image" src="https://github.com/princeton-nlp/SimCSE/assets/129758008/e5d0fe77-d2a5-4df1-8ea3-6a07b39d30f6">
above is to get the crop 10% data.

Then I run the run_unsup_example.sh with the train_file change and keep other parameters:
<img width="271" alt="image" src="https://github.com/princeton-nlp/SimCSE/assets/129758008/21937c88-398b-4015-84bd-c1e232e11aa2">

crop 10% data result is:
<img width="150" alt="image" src="https://github.com/princeton-nlp/SimCSE/assets/129758008/8234e306-a18a-4760-9d6a-af74dd57e860">

crop 20% data result is:
<img width="154" alt="image" src="https://github.com/princeton-nlp/SimCSE/assets/129758008/36a0d7be-9cbd-4196-9459-0afc4a70cc63">

Compared to the results in the table, I think these results is too high.  Is there any misunderstanding or operational error on my part?

I use pytorch1.9.0 and CUDA11.0

thanks! I would be appreciated if I could receive your reply!

**Comment by yaoxingcheng** [comment]
Hi. Thanks for your attention to our work. I believe there are some misunderstandings here. Given a train file ending with `txt`, where each line is one sentence, the code will still run the unsupervised version of SimCSE (same sentence forwarded twice with dropout as augmentation). To use explicit discrete augmentation, you need to store two randomly augmented sentences in a `csv` file with two columns, and use this `csv` file as the training file. You can read `train.py` from line 386 to see how data processing works in detail.

**Comment by tinaa4869** [comment]
> Hi. Thanks for your attention to our work. I believe there are some misunderstandings here. Given a train file ending with `txt`, where each line is one sentence, the code will still run the unsupervised version of SimCSE (same sentence forwarded twice with dropout as augmentation). To use explicit discrete augmentation, you need to store two randomly augmented sentences in a `csv` file with two columns, and use this `csv` file as the training file. You can read [`train.py`](https://github.com/princeton-nlp/SimCSE/blob/13361d0e29da1691e313a94f003e2ed1cfa97fef/train.py#L386) to see how data processing works in detail.
thanks for your reply!! follow your instruction, I think I get the appropriate results.
